packages/nfts/contracts/trailblazers-badges/TrailblazersBadges.sol

```diff
@@ -0,0 +1,190 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.24;
+
+import { ERC721EnumerableUpgradeable } from
+    "@openzeppelin/contracts-upgradeable/token/ERC721/extensions/ERC721EnumerableUpgradeable.sol";
+import { ECDSAWhitelist } from "./ECDSAWhitelist.sol";
+import { IMinimalBlacklist } from "@taiko/blacklist/IMinimalBlacklist.sol";
+import { Strings } from "@openzeppelin/contracts/utils/Strings.sol";
+
+contract TrailblazersBadges is ERC721EnumerableUpgradeable, ECDSAWhitelist {
+    /// @notice Base URI required to interact with IPFS
+    string private _baseURIExtended;
+    /// @notice Movement IDs
+    uint256 public constant MOVEMENT_NEUTRAL = 0;
+    uint256 public constant MOVEMENT_BASED = 1;
+    uint256 public constant MOVEMENT_BOOSTED = 2;
+    /// @notice Wallet-to-Movement mapping
+    mapping(address _user => uint256 _movement) public movements;
+    /// @notice Badge IDs
+    uint256 public constant BADGE_RAVERS = 0;
```

Comment:
constants should be put together, then follows the state variables.

`mapping(address _user => uint256 _movement) public movements;`
should be next to the other state variables.